src/control/server/storage/bdev/backend_class.go

```diff
@@ -116,36 +87,42 @@ func writeConf(log logging.Logger, templ string, req *FormatRequest) error {
 		req.OwnerUID, req.OwnerGID)
 }
 
+func writeJsonConfig(log logging.Logger, enableVmd bool, req *FormatRequest) error {
+	nsc, err := newSpdkConfig(log, enableVmd, req)
+	if err != nil {
+		return err
+	}
+
+	buf, err := json.MarshalIndent(nsc, "", "  ")
+	if err != nil {
+		return err
+	}
+
+	if err := writeConfFile(log, bytes.NewBuffer(buf), req); err != nil {
+		return err
+	}
+
+	return nil
+}
+
 // writeNvmeConf generates nvme config file for given bdev type to be consumed
 // by spdk.
 func (sb *spdkBackend) writeNvmeConfig(req *FormatRequest) error {
 	if req.ConfigPath == "" {
 		return errors.New("no output config directory set in request")
 	}
+	if req.Class == storage.BdevClassNvme && len(req.DeviceList) == 0 {
+		sb.log.Debug("skip write nvme conf for empty device list")
+		return nil
+	}
 
-	templ := map[storage.BdevClass]string{
-		storage.BdevClassNvme: clsNvmeTemplate,
-		storage.BdevClassKdev: clsKdevTemplate,
-		storage.BdevClassFile: clsFileTemplate,
-	}[req.Class]
-
-	// special handling for class nvme
-	if req.Class == storage.BdevClassNvme {
-		if len(req.DeviceList) == 0 {
-			sb.log.Debug("skip write nvme conf for empty device list")
-			return nil
-		}
-		if !sb.IsVMDDisabled() {
-			templ = `[Vmd]
-    Enable True
-
-` + templ
-		}
+	// TODO: enable AIO config support
+	if req.Class != storage.BdevClassNvme {
+		sb.log.Info("Skipping JSON SPDK config creation for non-NVMe bdev class")
```

Comment:
Is this coming in a subsequent commit to this PR?

Reply:
fixed